docs/guides/getting-started/hexagonal-architecture.md

```diff
@@ -0,0 +1,87 @@
+
+In contrast, while the layered architecture does allow for a clear and structured organization of the system, it can lead to stronger coupling between layers, potentially impacting flexibility and testability.
+
+### Drivers of Development
+
+In a layered architecture, where the persistence layer is the final layer and the domain is based on this, the design tends to be influenced by the underlying database structure. This is known as database-driven design, meaning that the organization and behavior of the software are determined by the structure and properties of the used database. While this can lead to efficient data processing, it can also cause the business logic to become tied to the database structure, potentially limiting system flexibility. Changes to the entities of the database thus necessitate an adjustment of the domain. The adjustment of this can lead to further necessary changes to dependent layers.
+
+On the other hand, the hexagonal architecture aims to put the focus on business logic and use cases, known as domain-driven design. This means that the software's design is driven by understanding and modeling the business domain, rather than by technical details such as the database structure. This approach can lead to software closely tailored to business requirements and easily adaptable to changes in these requirements.
+
+### Testability
+
+
+Regarding testability, the hexagonal architecture shows some advantages. By using ports and adapters, the individual components of the application can be isolated and tested independently. The strict decoupling between the application core and the outer components allows the use of so-called mock objects for testing, simulating interactions with the outer components. This can simplify testing and improve the quality of tests, as it allows for more precise control of specific scenarios and edge cases.
+
+While the layered architecture, though also capable of supporting well-structured and testable applications, can pose challenges concerning testability due to its stronger coupling between layers. Particularly, it may be more challenging to separate the layers for isolated testing. Changes to one layer may ripple to others, making testing more complicated.
+
+In summary, the hexagonal architecture tends to exhibit higher testability, mainly through its ability to isolate the individual components of the application.
+
+### Conclusion
+
+In conclusion, the hexagonal architecture appears to be superior in terms of coupling, drivers of development, and testability compared to the layered architecture. However, it's worth noting that the complexity of the hexagonal architecture might be perceived as unnecessary overhead in smaller systems. Nevertheless, when considering all factors, the hexagonal architecture provides a robust and flexible structure for software development, especially for projects requiring high testability and low coupling.
+
+
```

Comment:
💭 Maybe add a table highlighting the do's 🟢 and don's 🔴 
This way the reader gets some actionable take aways in the end.

Reply:
Should dos and donts look something like this?
Do:
Do use hexagonal architecture to enhance testability, especially for projects heavy on domain logic.

Don't:
Don't force hexagonal architecture into small or straightforward projects just for the sake of testability—it may introduce unnecessary complexity.

Reply:
I would go more into detail. Like, what are the rules you apply in an hexagonal architecture? 

Do's: 
- Keep your domain objects free from dependencies 
- Use Dependency Injection whenever possible
- ... 
Dont's: 
- Design your domain objects as they are represented in the database. 
- Allow bypassing of XYZ 

Wdyt?

Reply:
Sounds good to me 👍